Move theming support from Beta to Main

jordanjones243 opened the pull request:
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #95 

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)


## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

<!-- Generated by sourcery-ai[bot]: start summary -->

## Summary by Sourcery

Move theming support for the auro-badge component from beta to main, enabling customization through design tokens. Refactor the component to utilize a new dependency versioning system and enhance its functionality with new methods for handling SVG icons. Update documentation and tests to align with these changes, and clean up deprecated content.

New Features:
- Introduce theming support for the auro-badge component, allowing customization of styles using design tokens.

Enhancements:
- Refactor the auro-badge component to use a new dependency versioning system for auro-button and auro-icon elements.
- Improve the auro-badge component by adding a method to generate HTML elements with SVG icons.

Documentation:
- Update documentation to include new theming support details and modify API documentation paths.

Tests:
- Update tests to reflect changes in the auro-badge component, specifically targeting the new button element.

Chores:
- Remove deprecated CSS custom property fallbacks from the README and related documentation.

<!-- Generated by sourcery-ai[bot]: end summary -->

jason-capsule42 committed:
fix: update commitlint rules config

semantic-release-bot committed:
chore(release): 2.5.1-beta.1 [skip ci]

## [2.5.1-beta.1](https://github.com/AlaskaAirlines/auro-badge/compare/v2.5.0...v2.5.1-beta.1) (2024-07-27)

### Bug Fixes

* update commitlint rules config ([8f6c804](https://github.com/AlaskaAirlines/auro-badge/commit/8f6c804f56fb5aafa5db19a2274e84c1d7bc2eb4))

jason-capsule42 committed:
feat: add all code necessary for theme support

semantic-release-bot committed:
chore(release): 2.6.0-beta.1 [skip ci]

# [2.6.0-beta.1](https://github.com/AlaskaAirlines/auro-badge/compare/v2.5.1-beta.1...v2.6.0-beta.1) (2024-07-28)

### Features

* add all code necessary for theme support ([9332baf](https://github.com/AlaskaAirlines/auro-badge/commit/9332baf2dbbed563e540e4dad863913def5f8e7a))

jason-capsule42 committed:
feat: force version bump and release

semantic-release-bot committed:
chore(release): 2.6.0-beta.2 [skip ci]

# [2.6.0-beta.2](https://github.com/AlaskaAirlines/auro-badge/compare/v2.6.0-beta.1...v2.6.0-beta.2) (2024-07-28)

### Features

* force version bump and release ([6ecef1d](https://github.com/AlaskaAirlines/auro-badge/commit/6ecef1d8d6a3f326c747ddc948fc6bfb63cee8a2))

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/AlaskaAirlines/auro-badge?pullRequest=110) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/AlaskaAirlines/auro-badge?pullRequest=110) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: jason-capsule42<br/>:x: semantic-release-bot<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/AlaskaAirlines/auro-badge?pullRequest=110) it.</sub>

jordanjones243 commented on src/tokens.scss:
```diff
@@ -0,0 +1,8 @@
+
+:host {
+  --ds-auro-badge-border-color: var(--ds-color-border-subtle-inverse, #{$ds-color-border-subtle-inverse});
+  --ds-auro-badge-boxshadow-color: var(--ds-color-border-subtle-default, #{$ds-color-border-subtle-default});
```
Boxshaodw using default and border using inverse? Investigate more

jordanjones243 replied:
Boxshadow is using what I suspect to be the correct color.

jordanjones243 committed:
perf: update dependencies

BREAKING CHANGE: trigger major release for color theme support #95

jordanjones243 committed:
refactor: point to auro-library build scripts

jordanjones243 committed:
perf: refactor custom component registration config

jordanjones243 committed:
refactor: use auro-library component tag rename script

semantic-release-bot committed:
chore(release): 3.0.0-beta.1 [skip ci]

# [3.0.0-beta.1](https://github.com/AlaskaAirlines/auro-badge/compare/v2.6.0-beta.2...v3.0.0-beta.1) (2024-09-26)

### Performance Improvements

* refactor custom component registration config ([891f44a](https://github.com/AlaskaAirlines/auro-badge/commit/891f44a3a6945f96fbf2024875b76f16ae9eb8b8))
* update dependencies ([363d86d](https://github.com/AlaskaAirlines/auro-badge/commit/363d86d273adc3398f2c86d32ac0fa4db528a036))

### BREAKING CHANGES

* trigger major release for color theme support #95

jordanjones243 committed:
docs: remove beta releases from CHANGELOG

jordanjones243 commented on README.md:
```diff
@@ -82,15 +82,6 @@ The use of any Auro custom element has a dependency on the [Auro Design Tokens](
 
 <!-- AURO-GENERATED-CONTENT:END -->
 
-### CSS Custom Property fallbacks
-
-<!-- AURO-GENERATED-CONTENT:START (REMOTE:url=https://raw.githubusercontent.com/AlaskaAirlines/WC-Generator/master/componentDocs/partials/usage/cssFallbacks.md) -->
-[CSS custom properties](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties) are [not supported](https://auro.alaskaair.com/support/custom-properties) in older browsers. For this, fallback properties are pre-generated and included with the npm.
-
-Any update to the Auro Design Tokens will be immediately reflected with browsers that support CSS custom properties, legacy browsers will require updated components with pre-generated fallback properties.
-
-<!-- AURO-GENERATED-CONTENT:END -->
```
This has been done in every component.

jordanjones243 commented on src/auro-badge.js:
```diff
@@ -94,19 +111,59 @@ export class AuroBadge extends LitElement {
+
+    svg.setAttribute('slot', 'svg');
+
+    const iconHtml = html`<${this.iconTag} customColor customSize customSvg slot="icon">${svg}</${this.iconTag}>`;
+
+    return iconHtml;
```
Good idea to simplify, will address later.

blackfalcon commented:
:tada: This PR is included in version 3.0.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@aurodesignsystem/auro-badge/v/3.0.0)
- [GitHub release](https://github.com/AlaskaAirlines/auro-badge/releases/tag/v3.0.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
